packages/react/src/components/IonRouterContext.tsx

```diff
@@ -39,13 +39,17 @@ export const IonRouterContext = React.createContext<IonRouterContextState>({
  */
 export function useIonRouter(): UseIonRouterResult {
   const context = useContext(IonRouterContext);
-  return {
-    back: context.back,
-    push: context.push,
-    goBack: context.back,
-    canGoBack: context.canGoBack,
-    routeInfo: context.routeInfo,
-  };
+
+  return useMemo(
+    () => ({
+      back: context.back,
+      push: context.push,
+      goBack: context.back,
+      canGoBack: context.canGoBack,
+      routeInfo: context.routeInfo,
+    }),
+    [context.back, context.canGoBack, context.push, context.routeInfo]
```

Comment:
Sorry for the delay to respond! 

I can def reorder those for ya. I think since eslint just automagically put the deps in that it alphabetized them. 

I'll also look into why `context.back` didn't get added as well.

As for testing - I will def need an assist (or handoff) there. I'm super unfamiliar with testing suites and setup in general, hehe.

Thanks for reviewing!